quickwit-cli/src/lib.rs

```diff
@@ -654,6 +657,77 @@ impl ThroughputCalculator {
     }
 }
 
+/// Normalizes a URI
+/// A `file://` protocol is assumed if not specified.
+/// File URIs are resolved (normalised) to absolute path.
+/// Handles special path characters like (~, ., ..)
+pub fn normalize_uri(uri: &str) -> anyhow::Result<String> {
+    let parts: Vec<&str> = uri.split("://").collect();
+    let (protocol, mut path) = match parts.len() {
+        1 => (FILE_PROTOCOL, String::from(parts[0])),
+        _ => (
+            parts[0],
+            parts
+                .into_iter()
+                .skip(1)
+                .map(String::from)
+                .collect::<Vec<_>>()
+                .join(""),
+        ),
+    };
+
+    if protocol == FILE_PROTOCOL {
+        if path.starts_with('~') {
+            let home_dir_path = home::home_dir()
+                .context("Could nto get home directory.")?
+                .to_string_lossy()
+                .to_string();
+
+            path.replace_range(0..1, &home_dir_path);
+        }
+
+        if !path.starts_with('/') {
+            path.insert(0, '/');
+        }
+
+        path = normalize_path(Path::new(&path))
+            .to_string_lossy()
+            .to_string();
+    }
+
+    Ok(format!("{}://{}", protocol, path))
+}
+
+/// Normalizes a path without checking file existence in contrary to `Path::canonicalize`.
+/// https://github.com/rust-lang/cargo/blob/fede83ccf973457de319ba6fa0e36ead454d2e20/src/cargo/util/paths.rs#L61
+fn normalize_path(path: &Path) -> PathBuf {
```

Comment:
What is this for?
I don't quite understand...

Reply:
This does the same thing as `Path::canonicalize`, that is normalise the path by converting things like `/../.` into the actual path. But the version with Path::canonicalize will check if the file exist and blow otherwise. All I want here is a king of simple path normalization, whether the path is good or not will be resolved by the metastore or whatever happens to use it. I will elaborate this in the comment.

Reply:
Ah got it. No need to edit the comment. I was a bit slow there.